common/src/main/java/com/wynntils/models/discoveries/type/DiscoveryType.java

```diff
@@ -20,10 +20,9 @@ public enum DiscoveryType {
         this.color = color;
     }
 
-    public static DiscoveryType getDiscoveryTypeFromString(CodedString name) {
+    public static DiscoveryType getDiscoveryTypeFromString(StyledText name) {
         for (DiscoveryType type : values()) {
-            if (name.getInternalCodedStringRepresentation().charAt(1)
-                    == type.getColor().getChar()) {
+            if (name.getString().charAt(1) == type.getColor().getChar()) {
```

Comment:
I am usually against changes in a port PR.. However, name.firstPart.getStyle().getColor is probably better to compare here..

Reply:
maybe we should add a method for that ```java
name.getFirstPart().getPartStyle().getStyle().getColor().getValue() == type.getColor().getColor()```